core/templates/pages/topic-editor-page/navbar/topic-editor-navbar.component.ts

```
@@ -267,7 +269,9 @@ export class TopicEditorNavbarComponent {
     this.warningsAreShown = false;
     this.showTopicEditOptions = false;
     this.topic = this.topicEditorStateService.getTopic();
-    this.topicSkillIds = this.topic.getSkillIds();
+    if (this.topic) {
```

Suppose there is no topic at the time of init, and the topic loads later. Do we call ngOnInit() again (or some other method that initializes the topic-related stuff properly)?

---

Now I have added the code get skill ids whenever topic is initialized

---

But my question still stands. Let me try to break it down: the main question is, is it conceptually possible for there to be no topic at time of init?

If so, we need to initialize this.topic and this.topicSkillIds to null and [] respectively, and add a comment above them saying when exactly they'll be populated.

If not then we should throw an error if getTopic() returns null at this point.

---

@seanlip Yes, there is no topic at the time of init, infact when the component is getting rendered, the topic is not loaded from the backend yet, so inside ngOnInit(), this.topic is not defined. Only after the topic is fetched, and the onTopicInitialized event is emitted, is this.topic initialized with the actual topic value. But the tests written in its spec.ts file do not correspond with this new mechanism of initializing topic, which is now resulting in failing test cases on tryng `this.topic = null` inside ngOnInit(). So should I alter the tests in the spec.ts file accordingly?

---

Yes, you should alter the tests in the spec file accordingly -- see comment above.

---

Done